src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -108,7 +173,24 @@ public override BoundNode VisitInterpolatedString(BoundInterpolatedString node)
 
             BoundExpression? result;
 
-            if (CanLowerToStringConcatenation(node))
+            if (node.InterpolationData is not null)
+            {
+                // If we can lower to the builder pattern, do so.
+                var resultSymbol = _factory.SynthesizedLocal(_compilation.GetSpecialType(SpecialType.System_String), node.Syntax);
+                var resultTemp = _factory.Local(resultSymbol);
+
+                (ArrayBuilder<BoundExpression> builderPatternExpressions, BoundLocal builderTemp) = RewriteToInterpolatedStringBuilderPattern(node);
+
+                // resultTemp = builderTemp.ToStringAndClear();
+                var toStringAndClear = (MethodSymbol)Binder.GetWellKnownTypeMember(_compilation, WellKnownMember.System_Runtime_CompilerServices_InterpolatedStringBuilder__ToStringAndClear, _diagnostics, syntax: node.Syntax);
+                if (toStringAndClear is not null)
+                {
+                    builderPatternExpressions.Add(_factory.AssignmentExpression(resultTemp, BoundCall.Synthesized(node.Syntax, builderTemp, toStringAndClear)));
+                }
+
+                return _factory.Sequence(ImmutableArray.Create(resultSymbol, builderTemp.LocalSymbol), builderPatternExpressions.ToImmutableAndFree(), resultTemp);
```

Comment:
>resultTemp [](http://example.com/codeflow?start=151&length=10)

What is the motivation to use this temp? It looks like we could simply use the call at this position.